tests/integration/data_sources/test_local_files.py

```diff
@@ -32,3 +44,108 @@ def test_sample_dataset(self, local_files_dataset: Dataset) -> None:
         )
 
         assert len(features) == 2
+
+    @pytest.mark.parametrize(
+        "coordinate_mode", [GeojsonCoordinateMode.CRS, GeojsonCoordinateMode.WGS84]
+    )
+    def test_geojson_with_crs(
+        self, tmp_path: pathlib.Path, coordinate_mode: GeojsonCoordinateMode
+    ) -> None:
+        seattle = STGeometry(WGS84_PROJECTION, shapely.Point(-122.3, 47.6), None)
+        ds_path = UPath(tmp_path)
+
+        # This time we overwrite the GeoJSON with one in a different CRS and make sure
+        # it works no matter how the GeoJSON is encoded.
+        custom_projection = Projection(CRS.from_epsg(3857), 10, -10)
+        features = [
+            Feature(
+                geometry=seattle.to_projection(custom_projection),
+            ),
+        ]
+        src_data_dir = ds_path / "src_data"
+        src_data_dir.mkdir(parents=True, exist_ok=True)
+        vector_format = GeojsonVectorFormat(coordinate_mode=coordinate_mode)
+        vector_format.encode_vector(src_data_dir, custom_projection, features)
+
+        # Primary purpose of this test is to make sure LocalFiles can import the data,
```

Comment:
Suggestion: Move this part where you verify the encodding based on the desired coordinate mode to a separate integration test. It is clearer and easier to debug tests that only  have a single assert (or parametrized set of asserts. Also, consider having a different test for each coordinate mode as that will make things more readable.

Reply:
I'll remove this section, I guess it is already covered by the unit test.